src/contracts/core/PaymentCoordinator.sol

```diff
@@ -0,0 +1,502 @@
+        }
+        latestPaymentRangeTimestamp = uint64(block.timestamp);
+    }
+
+    /**
+     * @notice Claim payments against a given root (read from distributionRoots[claim.rootIndex])
+     * @param claim The PaymentMerkleClaim to be processed.
+     * Contains the root index, earner, payment leaves, and required proofs
+     * @dev only callable by the valid claimer, that is
+     * if claimerFor[claim.earner] is address(0) then only the earner can claim, otherwise only
+     * claimerFor[claim.earner] can claim the payments.
+     */
+    function processClaim(PaymentMerkleClaim calldata claim) external onlyWhenNotPaused(PAUSED_CLAIM_PAYMENTS) {
+        _processClaim(claim);
+    }
+
+    /**
+     * @notice Creates a new distribution root
+     * @param root The merkle root of the distribution
+     * @param paymentCalculationEndTimestamp The timestamp until which payments have been calculated
+     * @param activatedAt timestamp at which that the root can be claimed against
+     * @dev Only callable by the paymentUpdater
+     */
+    function submitRoot(
+        bytes32 root,
+        uint64 paymentCalculationEndTimestamp,
+        uint64 activatedAt
+    ) external onlyPaymentUpdater {
+        require(activatedAt >= block.timestamp, "PaymentCoordinator.submitRoot: activatedAt cannot be in the past");
+        activatedAt = activatedAt + activationDelay;
+        uint32 rootIndex = uint32(distributionRoots.length);
+        distributionRoots.push(
+            DistributionRoot({
+                root: root,
+                activatedAt: activatedAt,
+                paymentCalculationEndTimestamp: paymentCalculationEndTimestamp
+            })
+        );
+        emit DistributionRootSubmitted(rootIndex, root, paymentCalculationEndTimestamp, activatedAt);
+    }
+
+    /**
+     * @notice Create a RangePayment. Called from both `payForRange` and `payAllForRange`
+     */
+    function _payForRange(RangePayment calldata rangePayment) internal returns (bytes32) {
+        bytes32 rangePaymentHash = keccak256(abi.encode(msg.sender, paymentNonce, rangePayment));
+        require(
+            !isRangePaymentHash[msg.sender][rangePaymentHash],
+            "PaymentCoordinator._payForRange: range payment hash already submitted"
+        );
+        require(rangePayment.strategiesAndMultipliers.length > 0, "PaymentCoordinator._payForRange: no strategies set");
+        require(rangePayment.amount > 0, "PaymentCoordinator._payForRange: amount cannot be 0");
+        require(
+            rangePayment.duration <= MAX_PAYMENT_DURATION,
+            "PaymentCoordinator._payForRange: duration exceeds MAX_PAYMENT_DURATION"
+        );
+        require(
+            rangePayment.duration % calculationIntervalSeconds == 0,
+            "PaymentCoordinator._payForRange: duration must be a multiple of calculationIntervalSeconds"
+        );
+        if (retroactivePaymentsEnabled) {
```

Comment:
let's just take off this toggle imo

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/488/commits/a27911bceb4c60db7ad1a53551301dfb4615d020#diff-2a28246efa54c09dff5f0afef55f490d130b1a13c5b7e30efb51d5647c52a1a2L190